cpp/oneapi/dal/algo/svm/backend/gpu/train_kernel_thunder_dpc.cpp

```diff
@@ -55,10 +55,43 @@ static train_result call_daal_kernel(const context_gpu& ctx,
     auto arr_data  = row_accessor<const Float>{ data }.pull(queue);
     auto arr_label = row_accessor<const Float>{ labels }.pull(queue);
 
+    // TODO: make for dpcpp kernel
+    auto arr_new_label = array<Float>::empty(queue, row_count * 1);
+    arr_label.need_mutable_data(queue);
+    Float value_first_class_label  = arr_label[0];
+    Float value_second_class_label = arr_label[1];
+
+    arr_new_label[0] = Float(-1.0);
+    std::int64_t i   = 1;
+    for (; i < row_count; ++i) {
+        if (arr_label[i] == value_first_class_label) {
+            arr_new_label[i] = Float(-1.0);
+        }
+        else {
+            value_second_class_label = arr_label[i];
+            arr_new_label[i]         = Float(+1.0);
+        }
+    }
+    if (value_first_class_label == value_second_class_label) {
+        throw invalid_argument("Input label data should have more one unique label");
+    }
+
+    for (; i < row_count; ++i) {
+        if (arr_label[i] == value_first_class_label) {
+            arr_new_label[i] = Float(-1.0);
+        }
+        else if (arr_label[i] == value_second_class_label) {
+            arr_new_label[i] = Float(+1.0);
+        }
+        else {
+            throw invalid_argument("Input label data should have only two unique labels");
+        }
+    }
```

Comment:
You can use the method extracted earlier in CPU backend rather than copy-paste

Reply:
In the future I would like to rewrite this part here for a sycl. so I'm not sure what to merge the code.